File: apis/common/src/main/java/com/common/objects/models/api/ResponseModel.java
package com.common.objects.models.api;

import com.common.types.api.ResponseStatusType;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ResponseModel<T> {
    private T response;
    private String message;
    private ResponseStatusType responseStatus;
}

File: ticket-managment-system-web/src/app/core/models/api/response.model.ts
import { ResponseStatusType } from "../../enums";

export class ResponseModel<T> {
    public response: T | null = null;
    public message!: string;
    public responseStatus: ResponseStatusType = ResponseStatusType.Error;

    constructor(user: Partial<ResponseModel<T>>) {
        Object.assign(user);
    }
}